modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -121,7 +143,12 @@ public ScriptProcessor create(Map<String, Processor.Factory> registry, String pr
                 throw newConfigurationException(TYPE, processorTag, null, "Could not initialize script");
             }
 
-            return new ScriptProcessor(processorTag, script, scriptService);
+            CompiledScript compiledInlineScript = null;
+            if (script.getType() == ScriptType.INLINE) {
+                compiledInlineScript = scriptService.compile(script, ScriptContext.Standard.INGEST, script.getOptions());
```

Comment:
++ this is why I open WIP PRs! thanks – I'll check out what it means to resolve a script